src/MyEventEmitter.js

```diff
@@ -1,14 +1,72 @@
+/* eslint-disable max-len */
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.events = {};
+  }
+
+  checkEventName = (eventName) => {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+  };
+
+  on(eventName, listener) {
+    this.checkEventName(eventName);
+    this.events[eventName].push(listener);
+  }
+
+  once(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.on(eventName, onceListener);
+  }
```

Comment:
The `once` method currently removes the listener before it has been invoked. This could potentially lead to issues if an error is thrown during the invocation of the listener, as it would already have been removed. To fix this, you could invoke the listener before removing it.